index.js

```diff
@@ -32,21 +43,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var child_process = spawn(nativeBinary, args, {stdio: stdio});
-  child_process.on('close', function(exit) {
+  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
+  clangFormatProcess.on('close', function(exit) {
     if (exit) done(exit);
   });
-  return child_process;
+  return clangFormatProcess;
 }
 
-module.exports = clangFormat;
-module.exports.spawnClangFormat = spawnClangFormat;
-
-if (require.main === module) {
+function main() {
+  // Find clang-format in node_modules of the project of the .js file, or cwd.
+  var nonDashArgs = process.argv.filter(function(arg, idx) { return idx > 1 && arg[0] != '-' });
+  // Using the last file makes it less likely to collide with clang-format's argument parsing.
+  var lastFileArg = nonDashArgs[nonDashArgs.length - 1];
+  var basedir = lastFileArg ? path.dirname(lastFileArg) :  // relative to the last .js file given.
+                    process.cwd();                         // or relative to the cwd()
+  var resolvedClangFormat;
+  var clangFormatLocation;
   try {
-    spawnClangFormat(process.argv.slice(2), process.exit, 'inherit');
+    clangFormatLocation = resolve('clang-format', {basedir: basedir});
+    resolvedClangFormat = require(clangFormatLocation);
+  } catch (e) {
+    // Ignore and use the clang-format that came with this package.
+  }
+  var actualSpawnFn;
+  if (!resolvedClangFormat) {
+    actualSpawnFn = spawnClangFormat;
+  } else if (resolvedClangFormat.spawnClangFormat) {
+    actualSpawnFn = resolvedClangFormat.spawnClangFormat;
+  } else {
+    throw new Error('Incompatible clang-format loaded from ' + clangFormatLocation);
```

Comment:
yes - it doesn't have a `spawnClangFormat` method, so whatever `clang-format` was located doesn't have the right interface.